src/blob_gc_job_test.cc

```diff
@@ -580,7 +588,101 @@ TEST_F(BlobGCJobTest, LevelMergeGC) {
   DestroyDB();
 }
 
+TEST_F(BlobGCJobTest, RangeMergeScheduler) {
+  NewDB();
+  int max_sorted_run = 2;
+  std::vector<std::shared_ptr<BlobFileMeta>> files;
+
+  // one sorted run, no file will be marked
+  // run 1: [a, b] [c, d] [e, f] ... [p, q] [r, s]
+  for (int i = 0; i < 10; i++) {
+    auto file =
+        std::make_shared<BlobFileMeta>(i, 0, 0, 0, std::string('a' + i * 2, 1),
+                                       std::string('a' + i * 2 + 1, 1));
+    file->FileStateTransit(BlobFileMeta::FileEvent::kDbRestart);
+    files.emplace_back(file);
+  }
+  ScheduleRangeMerge(files, max_sorted_run);
+  for (const auto& file : files) {
+    ASSERT_EQ(file->file_state(), BlobFileMeta::FileState::kNormal);
+  }
+
+  // one more sorted run for 6 files, only those files will be marked
```

Comment:
please add a case that have one's end == other's start;
please add a case like this
```
run 1: [a             f]
run 2:    [b,      e]
run 3:      [c,d]
```
please add a case like this
```
run 1: [a, b] [c, d] [e, f] ... [n, o] [p, q] [r, s]
run 2:        [c, d] [e, f]            [p, q] [r, s]
```